tools/cbmc/proofs/utility/memory_assignments.c

```diff
@@ -0,0 +1,25 @@
+#define ensure_memory_is_valid( px ) (px != NULL)
+
+/* Implementation of safe malloc */
+void *safeMalloc(size_t xWantedSize) {
+	if(xWantedSize == 0) {
+		return NULL;
+	}
+	uint8_t byte;
+	return byte ? malloc(xWantedSize) : NULL;
```

Comment:
uint8_t byte;
    return byte ? malloc(xWantedSize) : NULL;
	
Maybe I miss something, but 'byte' is an uninitialised variable, used in an expression. Is that was is intended?
Hein

Reply:
For our proofs CBMC assumes an unconstrained uint8_t value for byte.